Warn against `targetSdkVersion` 21+ purely for multidex (#7188)
flutter/website: src/deployment/android.md

generated `FlutterMultiDexApplication` as the project's application.

{{site.alert.note}}

>   Multidex support is natively included when targeting min sdk 21+. However,
>   it is not recommended to target API 21+ purely to resolve the multidex issue
>   as this may inadvertently exclude users running older devices.

Reviewer: ```suggestion
  Multidex support is natively included when targeting Android SDK 21 or later.
  However, it isn't recommended to target API 21+ purely to resolve the multidex issue
  as this might inadvertently exclude users running older devices.
```
